File: day01/javascript/part01.js
const fs = require("fs");
const numbers = fs
  .readFileSync("input.txt", { encoding: "utf-8" })
  .split("\n")
  .filter((x) => !!x) //Boolean
  .map(Number);

let count = 0;
for (let i = 1; i < numbers.length; i++) {
  const previous = numbers[i - 1];
  const current = numbers[i];
  count += current > previous;
}
console.log(count); //1184

File: day01/python/part01.py
import os.path
with open(os.path.join(os.path.dirname(__file__), 'input.txt')) as f:
    lines = f.read()
    numbers = [int(line) for line in lines.splitlines()]

    # Method 1 
    # count = 0
    # prev=numbers[0]
    # for n in numbers[1:]:
    #     count += n>prev
    #     prev = n

    # Method 2
    # count = 0
    # for i in range(1, len(numbers)):
    #     count += numbers[i] > numbers[i-1]

    # Method 3
    # count = sum(numbers[i] > numbers[i-1] for i in range(1, len(numbers)))
    # print(count) #1184

    print(sum(a<b for a,b in zip(numbers,numbers[1:]))) #1184